cosmos/profiles/clickhouse/user_pass.py

```diff
@@ -0,0 +1,75 @@
+"""Maps Airflow Postgres connections using user + password authentication to dbt profiles."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from ..base import BaseProfileMapping
+
+
+class ClickhouseUserPasswordProfileMapping(BaseProfileMapping):
+    """
+    Maps Airflow generic connections using user + password authentication to dbt Clickhouse profiles.
+    https://docs.getdbt.com/docs/core/connect-data-platform/clickhouse-setup
+    """
+
+    airflow_connection_type: str = "generic"
+    dbt_profile_type: str = "clickhouse"
+    default_port = 9000
+    is_community = True
+
+    required_fields = [
+        "host",
+        "login",
+        "schema",
+        "clickhouse",
+    ]
+    secret_fields = [
+        "password",
+    ]
+    airflow_param_mapping = {
+        "host": "host",
+        "login": "login",
+        "password": "password",
+        "port": "port",
+        "schema": "schema",
+        "clickhouse": "extra.clickhouse",
+    }
+
+    @property
+    def profile(self) -> dict[str, Any | None]:
+        """Gets profile. The password is stored in an environment variable."""
+        profile_dict = {
+            **self.mapped_params,
+            **self.profile_args,
+            # password should always get set as env var
+            "password": self.get_env_var_format("password"),
+        }
+
+        if not profile_dict.get("driver"):
+            profile_dict["driver"] = "native"
+
+        if not profile_dict.get("port"):
+            profile_dict["port"] = self.default_port
+
+        if not profile_dict.get("secure"):
+            profile_dict["secure"] = False
+
+        return self.filter_null(profile_dict)
+
+    @property
+    def mock_profile(self) -> dict[str, Any | None]:
```

Comment:
Please, could we add a unit test covering this use-case?

Reply:
Added in commit: c8500324ad2c7d220dba3f6223de3c98bf84d1cd